### ISSUE TITLE
RetryHandlerOption incorrect max_delay default logic

### ISSUE THREAD

**Issue by pmcgaley** [microsoft/kiota-python#409]
There seems to be some confusion in this code between `delay` (defaults to 3 seconds) and `max_delay` (_should_ default to 180 seconds)

https://github.com/microsoft/kiota-python/blob/ad21998f4399a1d482b095478e247635a995708a/packages/http/httpx/kiota_http/middleware/options/retry_handler_option.py#L50

I'm hitting this using `GraphServiceClient` from `msgraph`. It's defaulting to `max_delay` of 3 seconds, resulting in limited or no retries.

Not sure if the intention is that you specify the max delay as `delay` in the constructor, or whether there should be `delay` and `max_delay` settings, but pretty sure it's not as intended currently.

**Comment by Ndiritu** [comment]
Hi @pmcgaley, thank you for reaching out.
The RetryHandler waits the number of seconds defined by the delay parameter. This defaults to `3 secs` and cannot be a value greater than `180 secs`.

I agree that the internal naming of the  variable within the class may be confusing, but the parameter name exposed via the constructor should communicate this.

You can change the default `delay` for the `GraphServiceClient` by customizing the HTTP client created:

```python
options = {}
options[RetryHandlerOption.get_key()] = new RetryHandlerOption(delay: 5)
http_client = GraphClientFactory.create_with_default_middleware(options=options)
request_adapter = GraphRequestAdapter(auth_provider, http_client)
client = GraphServiceClient(request_adapter=request_adapter)
```
This can also be done per request by passing the same options dictionary to the request configuration parameters in the GraphServiceClient's get()/post()... methods for each API path.

Please let me know if you need any further assistance.

**Comment by pmcgaley** [comment]
Hi. I don't think that's correct. The `RetryHandler` doesn't use that value as a wait time - it waits a time that's based either on `Retry-After` header, or a fallback of exponential backoff.

It stops retrying if the specified/computed retry time exceeds `max_delay`. And because of `self._max_delay: float = min(delay, self.MAX_DELAY)`, if I as in your example I specify a `delay` parameter of `5`, `max_delay` gets set to `5`. And then if the server responds with for example `Retry-After: 10`, that exceeds `max_delay` and no retry will be performed at all.

So basically the `delay` parameter actually specifies `max_delay`.

If I want it to retry for up to 180 seconds (which is theoretically the default) I actually have to specify `new RetryHandler(delay=180)`. If I leave it unspecified, `delay` defaults to `3`, and therefore because of the code above, `max_delay` is set to `3`.

Hope that makes sense.

**Comment by pmcgaley** [comment]
To be clear (because I realize I didn't say it explicitly above), I believe this is a bug.
